tests/src/JIT/Regression/JitBlue/GitHub_7508/Vector3Test.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+        float x, y, z;
+    }
+
+    public struct WrappedVector3
+    {
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public WrappedVector3( float x, float y, float z )
+        {
+            v = new System.Numerics.Vector3( x, y, z );
+        }
+
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        WrappedVector3( System.Numerics.Vector3 v )
+        {
+            this.v = v;
+        }
+
+        [MethodImpl( MethodImplOptions.AggressiveInlining )]
+        public static WrappedVector3 operator +( WrappedVector3 a, WrappedVector3 b )
+            => new WrappedVector3( a.v + b.v );
+
+        public float X
+        {
+            get { return v.X; }
+            set { v.X = value; }
+        }
+
+        public float Y
+        {
+            get { return v.Y; }
+            set { v.Y = value; }
+        }
+                
+        public float Z
+        {
+            get { return v.Z; }
+            set { v.Z = value; }
+        }
+
+        Vector3 v;
+    }
+
+    public class Program
+    {
+        static Random random = new Random( 12345 );
+        [MethodImpl( MethodImplOptions.NoInlining )]
+        static SimpleVector3 RandomSimpleVector3()
+            => new SimpleVector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static WrappedVector3 RandomWrappedVector3()
+            => new WrappedVector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static Vector3 RandomVector3()
+            => new Vector3( (float)random.NextDouble(), (float)random.NextDouble(), (float)random.NextDouble() );
+
+        public static void TestSimple()
+        {
+            var simpleA = RandomSimpleVector3();
+            var simpleB = RandomSimpleVector3();
+            var simpleC = simpleA + simpleB;
+            Console.WriteLine("Simple Vector3: {0},{1},{2}", simpleC.X, simpleC.Y, simpleC.Z);
+        }
+        public static void TestWrapped()
+        {
+            var wrappedA = RandomWrappedVector3();
+            var wrappedB = RandomWrappedVector3();
+            var wrappedC = wrappedA + wrappedB;
+            Console.WriteLine("Wrapped Vector3: {0},{1},{2}", wrappedC.X, wrappedC.Y, wrappedC.Z);
+        }
+        public static void TestSIMD()
+        {
+            var a = RandomVector3();
+            var b = RandomVector3();
+            var c = a + b;
+            Console.WriteLine("SIMD Vector3: {0},{1},{2}", c.X, c.Y, c.Z);
+        }
+        public static int Main( string[] args )
+        {
+            // First, test a simple (non-SIMD) Vector3 type
+            TestSimple();
+
+            // Now a wrapped SIMD Vector3.
+            TestWrapped();
+
+            // Now, SIMD Vector3
+            TestSIMD();
+
+            return 100;
+        }
```

Comment:
So this test only tests for JIT asserts? Otherwise, it always succeeds? Is there a way to verify correctness?